packages/devtools_app/lib/src/screens/debugger/debugger_controller.dart

```diff
@@ -34,21 +34,29 @@ class DebuggerController extends DisposableController
   // in `switchToIsolate`.
   DebuggerController({
     DevToolsRouterDelegate? routerDelegate,
-    this.initialSwitchToIsolate = true,
-  }) {
+    bool initialSwitchToIsolate = true,
+  }) : _initialSwitchToIsolate = initialSwitchToIsolate {
     autoDisposeStreamSubscription(
       serviceManager.onConnectionAvailable.listen(_handleConnectionAvailable),
     );
     if (routerDelegate != null) {
       codeViewController.subscribeToRouterEvents(routerDelegate);
     }
     if (serviceManager.hasService) {
-      initialize();
+      _initialize();
     }
   }
 
   final codeViewController = CodeViewController();
 
+  late final EvalService evalService = EvalService(
+    isolateRef: isolateRef,
+    variables: variables,
+    frameForEval: () => _frameForEval,
```

Comment:
_frameForEval should also be made a ValueListenable for consistency. That would probably cleanup some of the debugger UI that should be listening for changes to the frameForEval.

Reply:
I don't believe it makes sense to recalculate and update `frameForEval` every time `_selectedStackFrame` or `_stackFramesWithLocation` change, from both code readability and performance perspective.

Deleted getter and moved the logic to the function, as the getter is not used in other places..